### ISSUE TITLE
Feature Request: Line Breaks in Assignments in Compiled Javascript

### ISSUE THREAD

**Issue by btesser-r7** [Microsoft/TypeScript#7468]
One of typescript's key advantages over Babel (at least as far as I'm concerned), is the readability of its compiled javascript.  It is so readable, in fact, that I often use typescript for non-typescript projects, and just submit the compiled javascript.  One issue that often arises, however, is that the compiled javascript will not match style rules in these javascript projects.  One style rule is to have each assignment on its own line.  Typescript generally does this, except when destructuring.  It would be extremely valuable to be able to specify a compiler option for assignmentsOnNewLine which would add line breaks after each assignment.

For example, the following typescript code:
```typescript
const aVeryLargeObject = {
	firstValue: 1,
	secondValue: 2,
	thirdValue: 3,
	fourthValue: 4
}
const {firstValue, secondValue, thirdValue, fourthValue} = aVeryLargeObject;
```

Produces the following javascript:
```javascript
var aVeryLargeObject = {
    firstValue: 1,
    secondValue: 2,
    thirdValue: 3,
    fourthValue: 4
};
var firstValue = aVeryLargeObject.firstValue, secondValue = aVeryLargeObject.secondValue, thirdValue = aVeryLargeObject.thirdValue, fourthValue = aVeryLargeObject.fourthValue;
```

If this feature was implemented it should produce the following javascript:
```javascript
var aVeryLargeObject = {
    firstValue: 1,
    secondValue: 2,
    thirdValue: 3,
    fourthValue: 4
};
var firstValue = aVeryLargeObject.firstValue,
    secondValue = aVeryLargeObject.secondValue,
    thirdValue = aVeryLargeObject.thirdValue,
    fourthValue = aVeryLargeObject.fourthValue;
```

**Comment by RyanCavanaugh** [comment]
I think we could plausibly make this the default. Adding a commandline option for output formatting isn't going to happen, though -- there are plenty of good JS formatters available for reformatting the output to exact specifications.

**Comment by btesser-r7** [comment]
@RyanCavanaugh Fair enough.  It may be useful then to be able to specify a post-processor command when when using typescript's watch feature.
